### PR TITLE
Fix timeseries_dataset_from_array counts when sequence_stride > 1

### PR DESCRIPTION
**Pull request by basjacobs93**
Fixes https://github.com/keras-team/keras/issues/17392

### CODE REVIEW

**Comment by basjacobs93** [comment]
I believe the test `test_sampling_rate` (that I edited) was indeed also faulty, because if `data = np.arange(100)` (i.e. `0, 1, ..., 99`), then there are 84 such sequences of length 9 with sampling rate 2 possible, not 83:
```
0, 2, 4, 6, 8, 10, 12, 14, 16
1, 3, 5, 7, 9, 11, 13, 15, 17
2, 4, 6, 8, 10, 12, 14, 16, 18
...
81, 83, 85, 87, 89, 91, 93, 95, 97
82, 84, 86, 88, 90, 92, 94, 96, 98
83, 85, 87, 89, 91, 93, 95, 97, 99
```
which results in 17 batches of size 5, of which the last has 4 elements, not 3.
But please verify that my reasoning is correct.

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Thanks for the fix. Can u also update the example above at https://github.com/keras-team/keras/blob/16bc9f8b4b3cfc4d20f8ad683a8c400dcac5b35e/keras/utils/timeseries_dataset.py#L102. I think the last sequence of that example should be 
[81, 83, .... 99], and I believe the error of this example is also caused by the current issue. 

Thanks.

**Review by f1nn1711 — APPROVED** [review]

File: keras/utils/timeseries_dataset.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ def timeseries_dataset_from_array(
 
     # Determine the lowest dtype to store start positions (to lower memory
     # usage).
-    num_seqs = end_index - start_index - (sequence_length * sampling_rate) + 1
+    num_seqs = end_index - start_index - ((sequence_length - 1) * sampling_rate)
```
Comment (qlzh727) [inline keras/utils/timeseries_dataset.py:212]: the extra bracket for `(sequence_length - 1) * sampling_rate` can be skipped.

Comment (basjacobs93) [reply keras/utils/timeseries_dataset.py]: Okay, removed them in the new commit

**Review by qlzh727 — APPROVED** [review]
Thanks for the fix.

**Pull request merged**